File: VisualBasicAssignment/screen/kangtze/detail.vb
Imports System.Data.SqlClient
Imports System.Drawing.Image
Imports System.Drawing.Imaging
Imports System.Drawing.Bitmap
Imports System.IO

Public Class detail
    Dim ds As DataSet = New DataSet()
    Dim da As SqlDataAdapter

    Private Sub detail_Load(sender As Object, e As EventArgs) Handles MyBase.Load
        Dim cmdstr As String = "select * from food where food_id='" + detailid + "'"
        'PictureBox1.Image = Image.FromStream(ms)
        Dim arrImage() As Byte

        If openconnect() = True Then
            ds.Clear()
            da = New SqlDataAdapter(cmdstr, con)
            da.Fill(ds, "Food")

            If ds.Tables("Food").Rows.Count = 1 Then
                arrImage = ds.Tables("Food").Rows(0).Item(2)
                Dim ms As New MemoryStream(arrImage)
                Dim bitmap As New Bitmap(Image.FromStream(ms))

                lblName.Text = ds.Tables("Food").Rows(0).Item(1)
                lblDesc.Text = ds.Tables("Food").Rows(0).Item(3)
                lblPrice.Text = "RM" + ds.Tables("Food").Rows(0).Item(4).ToString

                pcbProduct.Image = bitmap
            End If
            closeconnect()
        End If
    End Sub

    Private Sub btnBack_Click(sender As Object, e As EventArgs) Handles btnBack.Click
        Me.Close()
    End Sub
End Class

File: VisualBasicAssignment/screen/kangtze/detail.cs
using System;
using System.Data;
using System.Data.SqlClient;
using Microsoft.VisualBasic.CompilerServices;

namespace VisualBasicAssignment
{
    public partial class detail
    {
        public detail()
        {
            InitializeComponent();
            _btnBack.Name = "btnBack";
        }

        private DataSet ds = new DataSet();
        private SqlDataAdapter da;

        private void detail_Load(object sender, EventArgs e)
        {
            using (var db = new FoodShopEntities1())
            {
                try
                {
                    var f = db.Foods.Find((object)Conversions.ToInteger(Module1.detailid));
                    if (f is null)
                    {
                    }
                    else
                    {
                        lblName.Text = f.title;
                        lblDesc.Text = f.descriptions;
                        lblPrice.Text = string.Format("RM {0:0.00}", f.price);
                        pcbProduct.Image = ExFunctions.byteToImage(f.image);
                    }
                }
                catch (Exception ex)
                {
                    Console.WriteLine("Exception caught: {0}", ex);
                }
            }
        }

        private void btnBack_Click(object sender, EventArgs e)
        {
            Close();
        }
    }
}